File: java/src/test/java/com/phenixrts/edgeauth/WhenVerifyingABadToken.java
package com.phenixrts.edgeauth;

import org.testng.Assert;
import org.testng.annotations.Test;

public class WhenVerifyingABadToken {
  private String token = "DIGEST:bad-token";

  @Test
  void theTokenFailsToVerify() {
    DigestTokens.VerifyAndDecodeResult result = new DigestTokens().verifyAndDecode("bad-secret", token);

    Assert.assertFalse(result.isVerified());
    Assert.assertEquals(result.getCode(), ECode.BAD_TOKEN);
    Assert.assertNull(result.getValue());
  }
}

File: dotnet/PhenixRTS.EdgeAuth.Tests/WhenVerifyingABadToken.cs
using Xunit;

namespace PhenixRTS.EdgeAuth.Tests
{
    public sealed class WhenVerifyingABadToken
    {
        private const string _token = "DIGEST:bad-token";

        [Fact]
        public void TheTokenFailsToVerify()
        {
            DigestTokens.VerifyAndDecodeResult result = new DigestTokens().VerifyAndDecode("bad-secret", _token);

            Assert.False(result.IsVerified());
            Assert.Equal(ECode.BAD_TOKEN, result.GetCode());
            Assert.Null(result.GetValue());
        }
    }
}
